pacwrapper.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"bytes"
+	"log"
+	"net/http"
+	"text/template"
+)
+
+type PACData struct {
+	Port     int
+	PACURL   string
+	Domain   string
+	Username string
+}
+
+type pacData struct {
+	PACData
+	PAC string
+}
+
+type PACWrapper struct {
+	data      pacData
+	tmpl      *template.Template
+	alpacaPAC string
+}
+
+// PACWrapper template for serving a PAC file to point at alpaca or DIRECT.
+// If we have a valid PAC file, we wrap that PAC file with a wrapper
+// function that only returns "DIRECT" or "localhost:port". If we do not
+// have a PAC file, the PAC function we serve only returns "DIRECT", which
+// should prevent all requests reaching us.
```

Comment:
If we're going to go with a 100 char line length, I think it'd be nice to wrap all the comments at 100 chars rather than 80 or whatever this is wrapped at.

Reply:
I wrap comments at 80 otherwise text becomes too hard to read. Or more that I don't let comment lines become too long - if they start at column 20, then I would be ok going to 100.

I didn't know 100 char line length had been decided on though. I have reformatted to 100 chars wide for the comment.

Done. Fixed in 46688aebb07260eac05dc4424dc9e8a3540650a9